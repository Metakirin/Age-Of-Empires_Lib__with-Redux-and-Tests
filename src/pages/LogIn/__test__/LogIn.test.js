import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { LogIn } from '..';

configure({ adapter: new Adapter() });

describe('Should render correctly "LogIn"', () => {
    const mockFn = jest.fn();

    const props = {
        handleLogin: mockFn,
    };
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LogIn {...props} />)
    });
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1);
    });
    it('component have children form', () => {
        expect(wrapper.find('form')).toHaveLength(1);
    });
    it('component have children "input"', () => {
        expect(wrapper.find('input')).toHaveLength(2);
    });
    it('component have children "Link"', () => {
        expect(wrapper.find('Link')).toHaveLength(1);
    });
    it('form should call onSubmit when submit', () => {
        wrapper.find('form').simulate('submit');
        expect(props.handleLogin).toHaveBeenCalledTimes(0);
    });
});