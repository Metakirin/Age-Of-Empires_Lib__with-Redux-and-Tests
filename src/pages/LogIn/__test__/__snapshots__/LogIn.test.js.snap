// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "LogIn" to Match Snapshot 1`] = `
<div
  className="login"
>
  <h1>
    Log In
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
       
      <span>
        Email
      </span>
      <input
        name="email"
        placeholder="email"
        type="email"
      />
    </label>
    <label>
       
      <span>
        Password
      </span>
      <input
        name="password"
        placeholder="password"
        type="password"
      />
    </label>
    <button
      className="form_btn"
      type="submit"
    >
      Login
    </button>
  </form>
  <p>
    or
  </p>
  <Link
    to="/signup"
  >
    Sign up
  </Link>
</div>
`;
